src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.AllLowerCamelCaseMatcher.cs

```diff
@@ -184,23 +207,36 @@ public AllLowerCamelCaseMatcher(string candidate, bool includeMatchedSpans, Stri
                     {
                         // We found a path that allowed us to match everything contiguously
                         // from the beginning.  This is the best match possible.  So we can
-                        // just stop now and return thie result.
-                        matchedSpans?.Insert(0, candidateMatchSpan);
-                        return (weight, matchedSpans);
+                        // just stop now and return this result.
+                        //
+                        // Return any previous best results back to the pool.
+                        matchedSpansInReverse?.Add(candidateMatchSpan);
```

Comment:
:bulb: This method could also benefit from block reordering like I mentioned above.